pkg/sql/logictest/testdata/logic_test/alter_table

```diff
@@ -440,9 +440,6 @@ INSERT INTO add_default (a) VALUES (3)
 statement error could not parse "foo" as type int
 ALTER TABLE add_default ALTER COLUMN b SET DEFAULT 'foo'
 
-statement error variable sub-expressions are not allowed in DEFAULT
-ALTER TABLE add_default ALTER COLUMN b SET DEFAULT $1
```

Comment:
Why remove these tests?

Reply:
I think they should be converted to `prepare a(int) as alter table ...`

Otherwise a naked $1 is not valid in a logic test.

Reply:
This was meant to test the "variable sub-expressions" error for alter table. Using placeholders without prepare is not valid, and now we get a different (earlier) error that's not specific to alter table. There are other testcases for the same error below though.

Reply:
👍 thanks for the explanation.